openid-connect-server/src/test/java/org/mitre/oauth2/service/impl/TestDefaultOAuth2AuthorizationCodeService

```diff
@@ -0,0 +1,53 @@
+package org.mitre.oauth2.service.impl;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+import org.junit.Test;
+import org.mitre.oauth2.model.AuthenticationHolderEntity;
+import org.mitre.oauth2.model.AuthorizationCodeEntity;
+import org.mitre.oauth2.repository.AuthorizationCodeRepository;
+import org.mitre.oauth2.repository.impl.JpaAuthenticationHolderRepository;
+import org.mitre.oauth2.repository.impl.JpaAuthorizationCodeRepository;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+
+public class TestDefaultOAuth2AuthorizationCodeService {
+
+	@Test
+	public void testConsumeAuthorizationCode()
+	{
+
+DefaultOAuth2AuthorizationCodeService defaultAuth = new DefaultOAuth2AuthorizationCodeService();
+AuthenticationHolderEntity authHolder = Mockito.mock(AuthenticationHolderEntity.class);
+OAuth2Authentication authentication = Mockito.mock(OAuth2Authentication.class);
+JpaAuthenticationHolderRepository authRepo = Mockito.mock(JpaAuthenticationHolderRepository.class);
+
+doNothing().when(authHolder).setAuthentication(Matchers.isA(OAuth2Authentication.class));
+when(authRepo.save(Matchers.isA(AuthenticationHolderEntity.class))).thenReturn(authHolder);
+AuthorizationCodeRepository repository = Mockito.mock(AuthorizationCodeRepository.class);
+AuthorizationCodeEntity entity = Mockito.mock(AuthorizationCodeEntity.class);
+when(repository.save(Matchers.isA(AuthorizationCodeEntity.class))).thenReturn(entity);
```

Comment:
This tells the repository mock to return the same entity for every "save" call, which doesn't really test what we want it to test. Instead, we're looking to make sure that the repository's "save" method is called with an authorization code with the right parameters -- specifically, with the associated authentication. This should be true even if we call it with separate authentications, which isn't happening here. 

Look into using JUnit/Mockito's "Answer" construct instead of straight mocks for this.